packages/atoms/src/Radio.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import {
+    NativeStyleObject,
+    NativeStyleUtils,
+    prepareNativeStyle,
+    useNativeStyles,
+} from '@trezor/styles';
+import { Pressable, View } from 'react-native';
+
+type RadioValue = string | number | undefined;
+
+interface RadioProps {
+    value: string;
+    isChecked?: boolean;
+    onPress?: (value: RadioValue) => void;
+    isDisabled?: boolean;
+    style?: NativeStyleObject;
+}
+
+type RadioStyleProps = {
+    isChecked?: boolean;
+    isDisabled?: boolean;
+};
+
+const getRadioBorderColor = (
+    utils: NativeStyleUtils,
+    { isChecked, isDisabled }: RadioStyleProps,
+) => {
+    if (isChecked && isDisabled) {
+        return {
+            borderColor: utils.colors.gray400,
+        };
+    }
+    if (isChecked) {
+        return {
+            borderColor: utils.colors.green,
+        };
+    }
+    return {
+        borderColor: utils.colors.gray400,
+    };
+};
+
+const radioStyle = prepareNativeStyle<RadioStyleProps>((utils, { isChecked, isDisabled }) => ({
+    height: 24,
+    width: 24,
+    backgroundColor: isDisabled ? utils.colors.gray400 : utils.colors.white,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderRadius: utils.borders.radii.round,
+    borderWidth: utils.borders.widths.md,
+    ...getRadioBorderColor(utils, { isChecked, isDisabled }),
+}));
+
+const radioCheckStyle = prepareNativeStyle<RadioStyleProps>((utils, { isDisabled }) => ({
+    height: 14,
+    width: 14,
+    borderRadius: utils.borders.radii.round,
+    backgroundColor: isDisabled ? utils.colors.gray600 : utils.colors.forest,
+}));
+
+export const Radio = ({ value, isChecked, onPress, isDisabled, style }: RadioProps) => {
+    const { applyStyle } = useNativeStyles();
+
+    const [innerIsChecked, setInnerIsChecked] = useState<boolean>(isChecked ?? false);
```

Comment:
Perhaps rename the prop to `isCheckedProp` so that the `inner` prefix is not needed?